packages/components/src/components/modals/DialogModal/DialogModal.tsx

```diff
@@ -10,11 +10,19 @@ const BodyHeading = styled.div`
     ${typography.titleMedium}
 `;
 
-const StyledIcon = styled(Icon)`
+const StyledIcon = styled(Icon)<{ iconVariant: 'success' | 'warning' }>`
```

Comment:
Can't you use just `<StyledIcon variant="warning />` etc.?

This way it's really hard to maintain it.